config/Config.cpp

```diff
@@ -0,0 +1,363 @@
+
+#include "Config.hpp"
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include <sstream>
+
+Config::Config(const std::string& config_file) 
+{
+	std::fstream configParser;
+	configParser.open(config_file);
+    if (!configParser.is_open()) {
+        throw std::runtime_error("config file open error");
+    }
+    parseLine(configParser);
+	while (!tokens.empty()) {
+		std::string token = tokens.front();
+		tokens.pop();
+
+		if (token == "http") {
+			if (tokens.front() == "{") {
+				tokens.pop();
+				parseHTTP();
+			} else {
+                throw std::out_of_range("missing '{' after http");
+            }
+		} else {
+			throw std::out_of_range("invalid config - constructor");
+		}
+	}
+}
+
+void Config::parseLine(std::fstream& configParser) {
+	std::string line;
+	while (std::getline(configParser, line)) {
+		size_t commentPos = line.find('#');
+		if (commentPos != std::string::npos) {
+			line = line.substr(0, commentPos);
+		}
+		line = trim(line);
+		if (line.empty())
+			continue;
+		if (line.back() != ';' && line.back() != '{' && line.back() != '}') {
+			throw std::out_of_range("invalid config - line");
+		}
+		tokenization(line);
+    }
+}
+
+std::string Config::trim(const std::string &str)
```

Comment:
멤버변수가 바뀌지 않으므로 const 함수로 만드는 것이 좋아보입니다.
또한 새로운 스트링을 만들어서 반환하는 것 보다는 str 자체를 조작하여 바꾸는 것이 새로운 할당이 없어서 효율적입니다.
```suggestion
void Config::trim(std::string &str) const
```

Reply:
수정했습니다!